client/src/Features/MicrosoftLearn/MicrosoftLearnFilterPane.tsx

```diff
@@ -0,0 +1,242 @@
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License.
+ *--------------------------------------------------------------------------------------------*/
+
+import React, { useEffect, useState } from 'react';
+import {
+  ActionButton,
+  DefaultButton,
+  Icon,
+  Panel,
+  PanelType,
+  PrimaryButton,
+  Separator,
+  styled,
+  Text
+} from '@fluentui/react';
+import { useStore } from '../../Stores/Core';
+import { useObserver } from 'mobx-react-lite';
+import { FilterComponent } from './MicrosoftLearnFilterComponentUtils';
+import { debounce } from 'lodash';
+import { IStylesOnly } from '../../Core/Utils/FluentUI/typings.fluent-ui';
+import { themedClassNames } from '../../Core/Utils/FluentUI';
+import { FilterType } from '../../Models/Learn/FilterType.model';
+import { FilterPaneStyles } from './MicrosoftLearnFilterPaneStyles';
+import { TAB_SCREEN_SIZE } from './MicrosoftLearnPage';
+
+const FilterPaneLarge = ({ styles }: IStylesOnly<FilterPaneStyles>): JSX.Element | null => {
+  const { filterStore } = useStore('microsoftLearnStore');
+  const classes = themedClassNames(styles);
+
+  const noFiltersExist = () => {
+    let selectedFilters = filterStore.selectedFilter;
+    return (
+      selectedFilters[FilterType.products].length === 0 &&
+      selectedFilters[FilterType.roles].length === 0 &&
+      selectedFilters[FilterType.types].length === 0 &&
+      selectedFilters[FilterType.levels].length === 0
+    );
+  };
+
+  return useObserver(() => {
+    return (
+      <div>
+        <ActionButton
+          onClick={() => filterStore.resetFilter()}
+          style={{ display: noFiltersExist() ? 'none' : 'block' }}
+          className={classes.clearAll}
+          text="Clear all filters"
+        />
+        <FilterComponent type={FilterType.products} name="Products" />
+        <Separator />
+        <FilterComponent type={FilterType.roles} name="Roles" />
+        <Separator />
+        <FilterComponent type={FilterType.levels} name="Levels" />
+        <Separator />
+        <FilterComponent type={FilterType.types} name="Types" />
+      </div>
+    );
+  });
+};
+
+const FilterPaneSmall = ({ styles }: IStylesOnly<FilterPaneStyles>): JSX.Element | null => {
+  const { filterStore, filteredCatalogContent, isLoadingCatalog } = useStore('microsoftLearnStore');
+
+  const [mainIsOpen, setMainOpen] = useState(false);
+  const [productIsOpen, setProductOpen] = useState(false);
+  const [roleIsOpen, setRoleOpen] = useState(false);
+  const [typeIsOpen, setTypeOpen] = useState(false);
+  const [levelIsOpen, setLevelOpen] = useState(false);
+  const [panelIsOpen, setPanelOpen] = useState(false);
+
+  const classes = themedClassNames(styles);
+
+  type PanelOptions = FilterType.products | FilterType.roles | FilterType.levels | FilterType.types | 'main';
+
+  const setOpen = (options: PanelOptions[]) => {
+    const open = (opt: PanelOptions) => options.includes(opt);
```

Comment:
How about doing it in the following way?
```suggestion
    const open = options.includes;
```